apm-sniffer/apm-sdk-plugin/httpClient-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/httpClient/v4/InstanceMehthodsInterceptPointImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.httpClient.v4;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static org.apache.skywalking.apm.plugin.httpClient.v4.Constants.INTERCEPT_CLASS;
+
+/**
+ * @author Alan Lau
+ */
+public class InstanceMehthodsInterceptPointImpl implements InstanceMethodsInterceptPoint {
```

Comment:
I am not sure for plugin case, this class is better than the old one, instrumentation defines the interceptor point, adding this could reduce the readability.

Reply:
Also, notice, nearly all interceptor points run in the old way. Don't provide a separate class.

Reply:
> > en, thanks for your reminder. this patch is not the code style at all.
> 
> I commented why I think it is a coding style thing.

May be you are right, everyone have different code style, what i think better is not really better to others. thank you a lot.